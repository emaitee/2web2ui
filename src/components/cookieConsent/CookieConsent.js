import React from 'react';
import { connect } from 'react-redux';
import { UnstyledLink, Snackbar } from '@sparkpost/matchbox';

import { giveConsent } from 'src/actions/cookieConsent';

import styles from './CookieConsent.module.scss';

export const ConsentBar = ({ onDismiss }) => <div className={styles.CookieConsent}>
  <div className={styles.ConsentBar}>
    <Snackbar maxWidth={600} onDismiss={onDismiss}>We use cookies to offer you a better browsing experience, analyze site traffic, and personalize content and advertisements.  Read more about how we use cookies and how you can control them by visiting our <UnstyledLink external to={'https://www.sparkpost.com/policies/privacy/'}>Cookie Policy</UnstyledLink>.  Note that parts of our site will not function properly if you disable them.  If you continue to use this site without disabling cookies, you consent to our use of them.</Snackbar>
  </div>
</div>;

export class CookieConsent extends React.Component {
  storeConsent = () => {
    this.props.giveConsent();
  };

  render() {
    const { consentGiven } = this.props;

    if (consentGiven) {
      return null;
    }

    return <ConsentBar onDismiss={this.storeConsent}/>;
  }
}

// connect consent state, set consent action
export default connect(
  ({ cookieConsent: { consentGiven }}) => ({ consentGiven }),
  { giveConsent }
)(CookieConsent);

