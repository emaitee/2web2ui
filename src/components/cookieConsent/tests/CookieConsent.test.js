import React from 'react';
import { shallow } from 'enzyme';
import { CookieConsent, ConsentBar } from '../CookieConsent';

describe('Component: CookieConsent', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      consentGiven: false,
      giveConsent: jest.fn()
    };

    wrapper = shallow(<CookieConsent {...props} />);
  });

  describe('ConsentBar', () => {
    it('should render correctly', () => {
      const props = { onDismiss: jest.fn() };
      expect(shallow(<ConsentBar {...props} />)).toMatchSnapshot();
    });
  });

  it('should render the banner without consent', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render the banner with consent', () => {
    wrapper.setProps({ consentGiven: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should give consent on dismissal', () => {
    wrapper.instance().storeConsent();
    expect(props.giveConsent).toHaveBeenCalledTimes(1);
  });
});
