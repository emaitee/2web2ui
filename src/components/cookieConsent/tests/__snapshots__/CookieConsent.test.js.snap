// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CookieConsent ConsentBar should render correctly 1`] = `
<div>
  <div>
    <Snackbar
      maxWidth={600}
      onDismiss={[MockFunction]}
      status="default"
    >
      We use cookies to offer you a better browsing experience, analyze site traffic, and personalize content and advertisements.  Read more about how we use cookies and how you can control them by visiting our 
      <UnstyledLink
        external={true}
        to="https://www.sparkpost.com/policies/privacy/"
      >
        Cookie Policy
      </UnstyledLink>
      .  Note that parts of our site will not function properly if you disable them.  If you continue to use this site without disabling cookies, you consent to our use of them.
    </Snackbar>
  </div>
</div>
`;

exports[`Component: CookieConsent should not render the banner with consent 1`] = `""`;

exports[`Component: CookieConsent should render the banner without consent 1`] = `
<Component
  onDismiss={[Function]}
/>
`;
